posts/formalizing-isocrystals-in-lean.md

```diff
@@ -0,0 +1,211 @@
+---
+author: 'Robert Y. Lewis, Heather Macbeth'
+category: 'New in mathlib'
+date: 2022-07-19 19:30:08 UTC+02:00
+description: ''
+has_math: true
+link: ''
+slug: classification-of-one-dimensional-isocrystals
+tags: ''
+title: Classification of one-dimensional isocrystals
+type: text
+---
+
+Last year, there was a [big mathlib refactor](https://leanprover-community.github.io/blog/posts/semilinear-maps/) to replace linear maps throughout the library with  *semilinear maps*,
+a more abstract concept which, importantly, unifies linear and conjugate-linear maps.
+
+But this is not the full extent of the generalization!  Our number theorist friends here in mathlib told us that we should
+make sure we chose this full generality of *semilinear* maps, maps $f:M \to N$ such that $f(ax)=\sigma(a)f(x)$ for some ring homomorphism $\sigma$ between the scalar rings of the modules $M$ and $N$.  So we and our coauthor Frédéric Dupuis implemented this full generality, and then asked them for an example to illustrate their need for this more abstract definition. This blog post tells the story of our little adventure in number theory.
+
+It turns out that the standard use of semilinear maps in number theory is for *Frobenius-semilinearity*, semilinearity with
+respect to the ring homomorphism of the fraction field of the $p$-typical Witt vectors over a perfect characteristic-$p$ integral domain which is induced by the Frobenius
+automorphism of that domain.  Let's backtrack to catch everyone up...
```

Comment:
Since this is the "teaser", you could consider saying something like
```suggestion
It turns out that the standard use of semilinear maps in number theory is for *Frobenius-semilinearity*,
semilinearity with respect to ring homomorphisms that are related to the Frobenius homomorphisms $x ↦ x^p$
that show up in the study of finite fields of characteristic $p$.
In our work, this Frobenius homomorphism manifested itself as an automorphism of
the fraction field of the $p$-typical Witt vectors over a perfect characteristic-$p$ integral domain
which is induced by the Frobenius automorphism of that domain.
Let's backtrack to catch everyone up...
```
That way the first reminder on Frobenius is maybe a little bit less intimidating.

Reply:
Why don't you even prefer "study of algebras over fields of characteristic $p$"? There is no good reason to stick to finite ones, and in particular most of the geometric instances come from infinite F_p-algebras, no?

Reply:
If you want to make it even less intimidating, you could simply write "It turns out that one standard use of semilinear maps in number theory is *Frobenius-semilinearity*, semilinearity with respect to the Frobenius endomorphism associated to a prime number p. The Frobenius endomorphism can be viewed as a non-archimedean analogue of complex conjugation, so perhaps one could argue that the two different motivations for formalising semilinear maps boil down to an archimedean and a nonarchimedean avatar of the same underlying idea.

Reply:
This is also a very nice way of putting it, yes.

Reply:
I think the stylistic point of this sentence before the teaser was to give a *very* small taste of what's coming, postponing the real details and exposition for later. It's supposed to be intimidating for effect. (Hence "let's backtrack...") So I'm not sure expanding it (beyond making sure it's technically correct) is the right move. Heather, what do you think?